### PR TITLE
Publish OpenAPI 3.0

### PR DESCRIPTION
**Pull request by hiddewie**
This PR migrates the swagger.json file from Swagger 2 to OpenAPI 3.0.

Using OpenAPI 3.0 instead of Swagger 2 integrates better with tooling, improves JSON schema support and allows expressing the requests/responses better.

With an OpenAPI 3.0 specification, tools like https://github.com/osmlab/maproulette-python-client can mostly be auto-generated and updated without all the API boilerplate.

A global security section allows configuring the API key to use for MapRoulette when trying the APIs.

Also, some issues in the specification file have been solved. In particular hundreds of invalid Swagger/OpenAPI specification fields have been fixed.

### Testing

OpenAPI validation:

https://apitools.dev/swagger-parser/online/# still shows a few hundred validation errors. Most have been solved already.

In a locally running Docker Swagger UI: 

image

API key: 
image

### CODE REVIEW

File: conf/v2_route/virtualchallenge.api [hunk 1]
```diff
@@ -123,28 +118,21 @@ GET     /virtualchallengebyname/:name                   @org.maproulette.control
...
-#     default: false
-#     type: boolean
-#   - name: apiKey
-#     in: header
-#     description: The user's apiKey to authorize the request
-#     required: true
-#     type: string
 ###
-DELETE  /virtualchallenge/:id                           @org.maproulette.controllers.api.VirtualChallengeController.delete(id:Long, immediate:Boolean ?= true)
+DELETE  /virtualchallenge/:id                           @org.maproulette.controllers.api.VirtualChallengeController.delete(id:Long, immediate:Boolean ?= false)
```
Comment (ljdelight) [inline conf/v2_route/virtualchallenge.api:122]: good catch

File: conf/swagger.yml [hunk 2]
```diff
@@ -27,275 +24,295 @@ tags:
...
-        $ref: "#/definitions/PublicUserSettings"
-        description: "Extra public user settings"
+          description: "The OSM User name"
+        created:
+          type: integer
+          format: epoch
+          description: "The date that the user was created in MapRoulette"
+        settings:
+          $ref: "#/components/schemas/PublicUserSettings"
+  securitySchemes:
```
Comment (ljdelight) [inline conf/swagger.yml:313]: Very cool

**Review by ljdelight — APPROVED** [review]
This is great, thank you for the patch!

**Comment by ljdelight** [comment]
This upgrade was on my todo list, thanks again 💯 
I'm going to rebase and merge as one since there are 20+ commits, copying your main PR comment for the commit body

**Comment by mvexel** [comment]
this is awesome, thanks so much @hiddewie !

@ mattmanley ping for Python client

**Comment by hiddewie** [comment]
Thanks for the feedback and quick reviews!

**Pull request merged**